ale_linters/erlang/erlc.vim

```diff
@@ -0,0 +1,88 @@
+" Author: Magnus Ottenklinger - https://github.com/evnu
+
+function! ale_linters#erlang#erlc#Handle(buffer, lines)
+  " Matches patterns like the following:
+  "
+  " error.erl:4: variable 'B' is unbound
+  " error.erl:3: Warning: function main/0 is unused
+  " error.erl:4: Warning: variable 'A' is unused
+  let l:pattern = '\v^([^:]+):(\d+): (Warning: )?(.+)$'
+
+  " parse_transforms are a special case. The error message does not indicate a location:
+  " error.erl: undefined parse transform 'some_parse_transform'
+  let l:pattern_parse_transform = '\v(undefined parse transform .*)$'
+  let l:output = []
+
+  let l:pattern_no_module_definition = '\v(no module definition)$'
+  let l:pattern_unused = '\v(.* is unused)$'
+
+  let l:is_hrl = expand('%:e') ==# 'hrl'
```

Comment:
This `expand` expression assumes that the buffer you are running this command on is the same buffer as the buffer you are checking. This might not be true. You should use the `a:buffer` variable to get the file extension for the buffer number.